### DISCUSSION TITLE
Documentation recommends proprietary PHP LSP (Intelephense) without disclosure

### DISCUSSION THREAD

**Discussion by gajdusek** [helix-editor/helix#8518] [General]
Hello,

[How to install the default language servers](https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers) suggests a proprietary and closed LSP for PHP: Intelephense, without even disclosing its proprietarity.

If users follow the recommended `npm install -g inelephense`  they unknowingly end with a proprietary software installed with a restrictive license they have never agreed upon and never got a chance to see, with a software hosted in private github repository.

It should suggest a FLOSS LSP or at least mention that this one is proprietary with the source code not available and a restrictive license to agree which is not displayed during installation.

Thank you, Petr

**Comment by the-mikedavis** [comment]
The Wiki is community editable - feel free to add a note about intelephense being proprietary and suggest a FLOSS language server there

**Comment by gajdusek** [comment]
Thanks for fast reply, didn't realize the wiki is community editable. I am in the process of finding a more free alternative and I will update the wiki with my findings.

**Comment by kirawi** [comment]
There is https://github.com/phpactor/phpactor

**Reply by mabasic** [reply]
I've added phpactor to the wiki here: https://github.com/helix-editor/helix/wiki/How-to-install-the-default-language-servers#phpactor

**Reply by webdev23** [reply]
Beside that, did you manage to make `phpactor` works with Helix at all?

When starting phpactor alone, it looks alright:
` [OK] Your system is ready to run the application. `

By checking `hx --health` after the configuration in the languages.toml, it looks alright too:
```
                          DAP              Highlight       TextObject     Indent
php         ✓ phpactor    None             ✓               ✓               ✓     
```
However, within Helix, editing a PHP file, if i do `gd` for example, i got:

`No configured language server supports goto definition`

Etc. LSP doesn't work, get called properly, and it's not a path issue. Any ideas?

**Reply by lens0021** [reply]
If you open helix with verbose mode(`hx -v`) and type `:log-open`, you can see the error log.
